### DISCUSSION TITLE
Is there any convinent way to sync data between unity and react?

### DISCUSSION THREAD

**Discussion by imwangji** [ReactUnity/core#51] [Q&A] [answered]
i'm new to unity and this library,please don't mind my stupid questions : I have 2 questions, please help me if you can:
1. I found I have to sync data between unity and react in this way below:
**unity**: add DataClass to Global Entries;
**react**: read Global Entrie, set data like this `globals.DemoManager.GetComponent("DemoManagerScript").username`
anyway, this sentense is a little bit verbose, is there any convinent way to do that?

2. I change data in Unity,but react does'nt change UI. what should I do? 

thanks for your help.

**Comment by gkurt** [comment, marked as answer]
Hey. Do not worry, your question is very valid and was asked many times. It is a topic that should be in the documentation but I didn't have time to do it yet.

1. Currently, this is the way to do it. If you want to make it shorter, you can assign it to a variable, or add `DemoManagerScript` directly to the globals, like this:

![gif](https://user-images.githubusercontent.com/6034931/131106539-2eb55a3e-ce23-4f50-852d-c3f007b3109a.gif)

2. If you want your changes to `Globals` to cause a rerender, you need to use `useGlobals`. 

However, this will only rerender on changes to the `Globals` itself, not its nested properties. For example, if you change `username` of `DemoManagerScript`, it won't cause a rerender. To do that, you can set it in Globals manually like `Globals["username"] = username`.

But there is also a mechanism to handle that. It is called watchables and it is similar to Globals. You can use it like:

C# side:

```
public class DemoManagerScript : MonoBehavior {
  public ReactUnity.Helpers.Watchable<string> username = "hello world";
  // .....
}
```

React side:

```
import { useGlobals, useWatchable } from '@reactunity/renderer';

function App() {
  var globals = useGlobals();
  var demoManager = globals.DemoManager.GetComponent("DemoManagerScript");
  var username = useWatchable(demoManager.username);

  // ....
}
```

**Reply by imwangji** [reply]
Thank you! hope this library become more and more popular,I'm trying to build my own game by ReactUnity.
